pkg/streamingpromql/operators/binary_operation.go

```diff
@@ -26,10 +26,12 @@ import (
 
 // BinaryOperation represents a binary operation between instant vectors such as "<expr> + <expr>" or "<expr> - <expr>".
 type BinaryOperation struct {
-	Left  types.InstantVectorOperator
-	Right types.InstantVectorOperator
-	Op    parser.ItemType
-	Pool  *pooling.LimitingPool
+	Left          types.InstantVectorOperator
+	Right         types.InstantVectorOperator
+	LeftIterator  types.InstantVectorSeriesDataIterator
+	RightIterator types.InstantVectorSeriesDataIterator
```

Comment:
I don't think these need to be exported, based on the convention we've been following elsewhere - they're an internal implementation detail of `BinaryOperation`.